### DISCUSSION TITLE
更新日志 - 20230217

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#597] [Announcements]
# 系统：
1. 提高内核兼容性，以便安装 OpenWRT 预编译的内核模块 （x86, rk33xx, rpi4等使用官方内核的设备）  #579 
2. 修复 luci 的 lua 版本的条件判断异常，增加错误字段高亮显示 https://github.com/linkease/doc.linkease.com/issues/81 https://github.com/openwrt/luci/issues/5705
3. 修复沙箱提交以后，overlayfs 可能出现冗余的 whiteout 标记的问题

# 设备：
* x86：
  1. 集成新 mt76 wifi 驱动，以支持 mt7921 的 AP 模式 （5G 频段有些信道无法启动 AP ，建议使用默认值）
  2. 集成 Intel AX201 wifi 驱动（不支持 AP 模式）
* R2S/R2C：
  1. 根据 PHY 芯片 ID 自动调整 MAC 的 delay 值，可能有助于提高网络稳定性 https://github.com/istoreos/istoreos/discussions/573#discussioncomment-4905249

**Comment by ks-bao** [comment]
嘿嘿，发布固件ing。。。

**Reply by samkuang1120** [reply]
升级后docker用不了。

Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: process_linux.go:508: setting cgroup config for procHooks process caused: bpf_prog_query(BPF_CGROUP_DEVICE) failed: function not implemented: unknown

**Reply by jjm2473** [reply]
> 升级后docker用不了。
> 
> Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: process_linux.go:508: setting cgroup config for procHooks process caused: bpf_prog_query(BPF_CGROUP_DEVICE) failed: function not implemented: unknown

@samkuang1120 你是在跑什么镜像的时候出现这个错误，或者是手动调整过容器的配置？我测试了hello-world镜像是没问题的

**Reply by ToutChenV587** [reply]
如何配置SSL访问，之前lede可以设置，istoreos没找到地方设置😊

**Reply by jjm2473** [reply]
@ToutChenV587 在istore安装uhttpd，就有界面设置ssl了

**Reply by wushuo894** [reply]
我的uhttpd一设置 0.0.0.0:443就报错，系统是新装的

**Reply by jjm2473** [reply]
@W2725730722 不需要设置地址啊，只要勾选“将所有 HTTP 重定向到 HTTPS”就行了。我发现luci有bug，判断ipv6错误，所以要删掉两个ipv6的监听地址才保存，也就是删掉“[::]:80”和“[::]:443”

**Reply by wushuo894** [reply]
> @W2725730722 不需要设置地址啊，只要勾选“将所有 HTTP 重定向到 HTTPS”就行了。我发现luci有bug，判断ipv6错误，所以要删掉两个ipv6的监听地址才保存，也就是删掉“[::]:80”和“[::]:443”

原来如此，我回家试一下

**Comment by dunquzhiyi** [comment]


**Reply by sleikang** [reply]
我的也是这样 这个还是前面还是升级前能启动的，我删除后从小下载容器就这样了
Containers: start mynodered...code:400 OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: process_linux.go:508: setting cgroup config for procHooks process caused: open /sys/fs/cgroup/docker/cab616035f3a5b70836aa3b9580724418efebbc49b6c1631049dc76957b09bd9/io.weight: no such file or directory: unknown

**Reply by sleikang** [reply]
机子是X86版本

**Comment by jjm2473** [comment]
已确认rk3568方案机型固件的docker容器无法启动，会尽快修复。

**Reply by ks-bao** [reply]
> 已确认rk3568方案机型固件的docker容器无法启动，会尽快修复。

固件已经更新。

**Reply by samkuang1120** [reply]
更新完， 可以用了， 谢谢

**Reply by sleikang** [reply]
怎么更新 x86的固件也这样

**Reply by jjm2473** [reply]
x86固件有问题吗，把日志发出来看看。我测试了hello-world容器可以正常运行

**Reply by sleikang** [reply]
> x86固件有问题吗，把日志发出来看看。我测试了hello-world容器可以正常运行

自从更新0217后就添加新的容器都和上面一样的报错，试了几个容器都是，我开始以为是容器不支持，后面我把0217系统以前装上的node-red删除了从新下载都报下面这种错误
Containers: start mynodered...code:400 OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: process_linux.go:508: setting cgroup config for procHooks process caused: open /sys/fs/cgroup/docker/cab616035f3a5b70836aa3b9580724418efebbc49b6c1631049dc76957b09bd9/io.weight: no such file or directory: unknown

**Reply by jjm2473** [reply]
@sleikang 把你的部署的命令发一下，我看看新固件修复没

**Reply by jjm2473** [reply]
我在开发版固件用官方的这个方式部署没问题 https://nodered.org/docs/getting-started/docker
下个版本应该也能修复你的问题

**Reply by sleikang** [reply]
> @sleikang 把你的部署的命令发一下，我看看新固件修复没

 docker run -it -p 1880:1880 -v node_red_data:/data --name mynodered nodered/node-red
我降级了固件就可以了 17号那个版本如果是新添加的容器都会启动不了，已经启动的容器就可以启动，已经启动的容器删除和删除镜像都会再次添加启动不了

**Comment by oofyuan** [comment]
希望在X86上 增加对8156B 2.5G USB网卡的支持

**Reply by jjm2473** [reply]
难道现在不支持？

**Reply by jjm2473** [reply]
`lsmod | grep r8152` 不是有驱动吗

**Reply by oofyuan** [reply]
诶那奇怪了。我的8156B也显示连接。但是插上网线之后，没有握手速度。这是为什么？

**Reply by oofyuan** [reply]
网卡肯定没有问题，因为我现在把他插在iKuai上在用。2.5G握手成功

**Reply by jjm2473** [reply]
哦，我看r8152驱动还是内核自带的，而且没有打补丁，不支持r8156方案。之后移植新的驱动就好了

**Reply by oofyuan** [reply]
感谢回复。希望早日支持8156B这个便宜可行的方案！

**Reply by jjm2473** [reply]
以前就给老内核移植过r8152，只是没注意到5.4内核的r8152也不能用，而且没人反馈

**Reply by oofyuan** [reply]
60元就可以拓展一个2.5G的网口还是很香的。

**Comment by lmxslpc** [comment]
我把uhttpd的默认端口80,443改掉了,然后把heimdall容器的80:80/tcp和443:443/tcp映出来,可是就是不能通过80,443端口访问heimdall,映射别的端口都可以访问,有大佬能告诉我怎么回事么

**Reply by jjm2473** [reply]
一般都是运营商的问题

**Reply by lmxslpc** [reply]
> 一般都是运营商的问题

是家里的路由   用内网ip访问的

**Reply by jjm2473** [reply]
那换个浏览器试试

**Reply by lmxslpc** [reply]
> 那换个浏览器试试

搞定了 谢谢 就是浏览器缓存的问题

**Comment by picc745514** [comment]
R2C更新后，WAN如果是百兆，工作不正常，接口页面显示只有发送数据，接收数据为0，物理状态正常。
更换千兆后正常，两个网口Homebox测速依旧惨不忍睹

**Comment by fanzhong-cn** [comment]
特地买了一个7921，没成，AP摆弄不出来，技不如人。

**Reply by jjm2473** [reply]
pcie接口的吗？

**Reply by ghost** [reply]
才看到这个更新，请问R71S可以安装外置天线吗？

**Reply by jjm2473** [reply]
安装天线跟固件有啥关系？

**Comment by JasperHu95** [comment]
我通过外置的硬盘盒将两块3.5寸的硬盘连接到物理机上，无法读出SMART信息，而我这样在unraid系统上操作时，是可以读取到SMART信息的。

**Reply by jjm2473** [reply]
如果使用UAS协议，是可以读到smart信息的，但有些硬盘盒兼容性不好，会被linux降级成UMS，就没有smart信息了。

**Reply by JasperHu95** [reply]
> 如果使用UAS协议，是可以读到smart信息的，但有些硬盘盒兼容性不好，会被linux降级成UMS，就没有smart信息了。
感谢您的回答，那么您可否推荐一款硬盘盒？或者，有无一种可能，USB接口会对此产生影响？

**Reply by jjm2473** [reply]
没推荐，最好用sata接口。usb硬盘盒要自己试才知道
